src/Illuminate/Queue/Worker.php

```diff
@@ -690,6 +712,33 @@ public function setCache(CacheContract $cache)
         $this->cache = $cache;
     }
 
+    /**
+     * Set the name of the worker.
+     *
+     * @param  string $name
+     * @return void
+     */
+    public function setName($name)
+    {
+        $this->name = $name;
+    }
+
+    /**
+     * Register a callback to be executed to pick jobs.
+     *
+     * @param  string  $workerName
+     * @param  callable  $callback
+     * @return void
+     */
+    public static function pickJobsUsing($workerName, $callback)
+    {
+        if (is_null($callback)) {
+            static::$jobPickerCallbacks = [];
```

Comment:
Shouldn't this just clear the callback for the selected job, currently this removes all callbacks.